tf_encrypted/keras/backend.py

```diff
@@ -0,0 +1,59 @@
+"""TFE Keras backend. Most of the code was
+borrowed from the tf.keras codebase.
+"""
+import threading
+
+from tensorflow.python.framework import ops
+
+import tf_encrypted as tfe
+
+# This is a thread local object that will hold the default internal TFE session
+# used by TFE Keras. It can be set manually via `set_session(sess)`.
+_SESSION = threading.local()
+
+
+def get_session(op_input_list=()):
+  """Returns the session object for the current thread."""
+  global _SESSION
+  if getattr(_SESSION, 'session', None) is not None:
+    return _SESSION.session
+  default_session = ops.get_default_session()
+  if default_session is not None:
+    # If the default session is a TFE Session return this session
+    if isinstance(default_session, tfe.Session()):
+      return default_session
+    if not isinstance(default_session, tfe.Session()):
+      raise TypeError(
```

Comment:
what's the reason for this? it seems that in most situations the user would have to manually set a session first for tfe.keras to work -- maybe this is expected behaviour..?

Reply:
I have added this error in case, the model definition or set_weights is ran under `tf.Session()` instead of tfe.Session().